crates/pallet-domains/src/lib.rs

```diff
@@ -755,11 +853,71 @@ mod pallet {
+                        genesis_domain.runtime_name.clone(),
+                        genesis_domain.runtime_type.clone(),
+                        genesis_domain.runtime_version.clone(),
+                        genesis_domain.code.clone(),
+                        Zero::zero(),
+                    )
+                    .expect("Genesis runtime registration must always succeed");
+
+                    // Instantiate the genesis domain
+                    let domain_config = DomainConfig::from_genesis::<T>(genesis_domain, runtime_id);
+                    let domain_owner = genesis_domain.owner_account_id.clone();
+                    let domain_id = do_instantiate_domain::<T>(
+                        domain_config,
+                        domain_owner.clone(),
+                        Zero::zero(),
+                    )
+                    .expect("Genesis domain instantiation must always succeed");
+
+                    // Register domain_owner as the genesis operator.
+                    let operator_config = OperatorConfig {
+                        signing_key: genesis_domain.signing_key.clone(),
+                        minimum_nominator_stake: genesis_domain
+                            .minimum_nominator_stake
+                            .saturated_into(),
+                        nomination_tax: genesis_domain.nomination_tax,
+                    };
+                    let operator_stake = T::MinOperatorStake::get();
+                    let operator_id = do_register_operator::<T>(
+                        domain_owner,
+                        domain_id,
+                        operator_stake,
+                        operator_config,
+                    )
+                    .expect("Genesis operator registration must succeed");
+
+                    // TODO: Enact the epoch transition logic properly.
+                    Operators::<T>::mutate(operator_id, |maybe_operator| {
+                        let operator = maybe_operator
+                            .as_mut()
+                            .expect("Genesis operator must exist");
+                        operator.current_total_stake = operator_stake;
+                    });
+                    DomainStakingSummary::<T>::insert(
+                        domain_id,
+                        StakingSummary {
+                            current_epoch_index: 0,
+                            current_total_stake: operator_stake,
+                            current_operators: vec![operator_id],
+                            next_operators: vec![],
+                        },
+                    );
+                }
+            }
 
             do_upgrade_runtimes::<T>(block_number);
 
-            Weight::zero()
+            let results = SuccessfulBundles::<T>::clear(u32::MAX, None);
+            let db_weight = T::DbWeight::get();
+            db_weight
+                .reads(results.loops as u64)
+                .saturating_add(db_weight.writes(1))
```

Comment:
This can be simply ignored actually for now as the weights for the entire function are still incorrect even with this effort.